### DISCUSSION TITLE
Warn from WorkManager

### DISCUSSION THREAD

**Discussion by radoslawtwardy** [confluentinc/parallel-consumer#80] [Q&A] [answered]
Sometimes i have a lot of warns:
`17:19:47.392 [control] WARN  i.c.parallelconsumer.WorkManager - Work (WorkContainer(topicName-1:10:fed5ca5a-e4de-4229-b42c-0b9ebddfa272)) has delay passed? (true) or is NOT in flight? (false, time spent in execution queue: PT15.002S), hasNotAlreadyPreviouslySucceeded? false can't take...
`

Could you explain me that?

**Comment by astubbs** [comment, marked as answer]
Yeah I need to clean that up. It's a debug level statement for a situation I didn't think would arise much, and if it did it's a potential problem - and only prints at WARN if the record has been waiting in the queue for over 10 seconds to process.

Can you give me some more context around what was happening when this happened? How loaded was the system?

It's this line here: https://github.com/confluentinc/parallel-consumer/blob/5fa5f59b96b02813b10ed75ee1533b58b829388c/parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/WorkManager.java#L444

I've pushed this minor clean up here: https://github.com/confluentinc/parallel-consumer/compare/master...astubbs:logging-cleanup

Keen to hear your thoughts..

**Reply by radoslawtwardy** [reply]
Thanks for reply. It explains everything. I do some tests - i definitely it was overloaded in my case.
